### PR TITLE
Add blackbox test using test vectors created with lesspass-cli.

### PR DESCRIPTION
**Pull request by teythoon**

### CODE REVIEW

File: tests/blackbox.rs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+//!                                                        --${SYMBOLS}symbols \
+//!                                                    | sed -e 's/\\/\\\\\\\\/g' -e 's/"/\\"/g' )\""
+//!                                     echo -n ", "
+//!                                 done
+//!                             done
+//!                         done
+//!                     done
+//!                     echo "]);"
+//!                     echo "}\n"
+//!                     TEST="$(expr $TEST + 1)"
```
Comment (71) [inline tests/blackbox.rs]:
```suggestion
//!                     TEST="$((TEST + 1))"
```

File: tests/blackbox.rs [hunk 2]
```diff
@@ -0,0 +1,283 @@
+//! Tests lesspass.rs using lesspass-cli as oracle.
+//!
+//! The test vectors in this file are generated by lesspass-cli using
+//! the following script.
+//!
+//! ```sh
```
Comment (71) [inline tests/blackbox.rs]: Do you mind putting the script in its own file, and instead linking to that file in the comment?

Additionally, just for a "prettier" `cargo test` output, I made two small changes below.

File: tests/blackbox.rs [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+//!     exit 2
+//! fi
+//!
+//! TEST=0
+//! for SITE in example.org example.net; do
+//!     for LOGIN in user@example.org user; do
+//!         for PASSWORD in password foobar; do
+//!             for COUNTER in 1 2; do
+//!                 for LENGTH in 12 16; do
+//!                     echo "#[test]\nfn vectors_$TEST() {"
```
Comment (71) [inline tests/blackbox.rs]:
```suggestion
//!                     echo "#[test]\nfn vectors_$(printf "%02d" $TEST)() {"
```

**Review by 71 — CHANGES_REQUESTED** [review]
Thanks for the tests, they're especially useful for your next PR.

File: tests/blackbox.rs (hunk 1 above)
Comment (teythoon) [reply tests/blackbox.rs]: Ah nice, I thought that was a bashism.

File: tests/blackbox.rs (hunk 2 above)
Comment (teythoon) [reply tests/blackbox.rs]: Sure thing.

**Pull request merged**